### PR TITLE
fix(public_spec): check exports of barrels instead of angular2/angular2

### PR DESCRIPTION
**Pull request by jeffbcross**
This changes the public api spec to check each public barrel individually
to make sure its API has not changed. The previous API spec has been
preserved but split into respective barrels.

The compiler barrel has been added to the spec, along with all of its
public exports. Previously, angular2/angular2 was only exporting a
handful of symbols from compiler, so there are now many more symbols
being tested in the spec for compiler than previously.

Part of #5710

CC @IgorMinar

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@jeffbcross thnx for the work on this! It is kind of hard to review since GitHub doesn't show diff for the test (probably the diff gets too big for its taste...). From the brief look changes seem reasonable for me but:
* TravisCI is not happy (Dart part) - looks like a legit failure 
* Would be good to have another look from one of the "Public API masters"

But yeh, I guess sorting out Dart needs to happen first

**Comment by jeffbcross** [comment]
@pkozlowski-opensource Dart is an easy fix (unless there are more problems beyond the initial obvious problem :))

Yeah, the diff is hard. Here's what I can tell you about my process to ensure that the list is correct.

 * Every API that was in `NG_ALL` is in one of the other lists; nothing was removed.
 * I looked at angular2.ts to determine which barrels should be included in this spec.
 * For the `compiler` barrel, I moved the APIs that were exported from angular2.ts to the new compiler barrel list, ran the test, then added all the remaining compiler barrel exported APIs to the spec.

So nothing was removed, but `compiler` APIs were added.

I'll get one of the Guardians of the API™ to review as well.

**Comment by pkozlowski-opensource** [comment]
Sounds good @jeffbcross :+1:

**Comment by jeffbcross** [comment]
Correction: Dart does have more problems :). I need to figure out how to deal with `ngUpgrade` import.

File: modules/angular2/test/symbol_inspector/symbol_inspector.dart [hunk 1]
```diff
@@ -14,14 +18,17 @@ const IGNORE = const {
...
+  'ngInstrumentation': 'angular2.instrumentation',
+  'ngPlatformBrowser': 'angular2.platform.browser'
 };
 
-// Have this list here to trick dart to force import.
-var libs = [simple_library.A, angular2.Component];
-
 List<String> getSymbolsFromLibrary(String name) {
+  // Have this list here to trick dart to force import.
+  var libs = [simple_library.A, Component];
```
Comment (jeffbcross) [inline modules/angular2/test/symbol_inspector/symbol_inspector.dart]: Move this back

**Comment by jeffbcross** [comment]
Got verbal LGTM from @tbosch with latest changes, once I make minor fix to `symbol_inspector.dart`.

@pkozlowski-opensource FYI

**Comment by pkozlowski-opensource** [comment]
Awesome! As soon as this one gets in we need to merge https://github.com/angular/angular/pull/5828 and see what happens :-)

File: modules/angular2/test/symbol_inspector/symbol_inspector.dart (hunk 1 above)
Comment (jeffbcross) [inline modules/angular2/test/symbol_inspector/symbol_inspector.dart]: Nvm

**Comment by jeffbcross** [comment]
@tbosch is taking over fixing `symbol_inspector` to get rid of the DartAnalyzer unused import errors.

**Pull request closed without merging**